### PR TITLE
paho-mqtt-cpp: fix CMake imported target

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **paho-mqtt-cpp/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/paho-mqtt-cpp/all/conanfile.py [hunk 1]
```diff
@@ -83,6 +83,9 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
-        self.cpp_info.libs = tools.collect_libs(self)
         self.cpp_info.names["cmake_find_package"] = "PahoMqttCpp"
         self.cpp_info.names["cmake_find_package_multi"] = "PahoMqttCpp"
+        cmake_target = "paho-mqttpp3" if self.options.shared else "paho-mqttpp3-static"
+        self.cpp_info.components["_paho-mqtt-cpp"].names["cmake_find_package"] = cmake_target
+        self.cpp_info.components["_paho-mqtt-cpp"].names["cmake_find_package_multi"] = cmake_target
+        self.cpp_info.components["_paho-mqtt-cpp"].libs = tools.collect_libs(self)
```
Comment (gocarlos) [inline recipes/paho-mqtt-cpp/all/conanfile.py:91]: Does this mean that i need to update my cmake scripts depending on if I’m building (or using) a shared to static library?

Comment (SpaceIm) [reply recipes/paho-mqtt-cpp/all/conanfile.py:91]: Yes, if you want transparent cmake integration.
It's worth noting, that `cmake_find_package` and `cmake_find_package_multi` still create `PahoMqttCpp::PahoMqttCpp` (unofficial) target.

Comment (gocarlos) [reply recipes/paho-mqtt-cpp/all/conanfile.py:91]: poh, its a pain that people (upstream) sill doing this... they could also start creating targets depending on the compiler, standard, with and without ssl....

Comment (SpaceIm) [reply recipes/paho-mqtt-cpp/all/conanfile.py:91]: Loot at `paho-mqtt-c` and cry :sweat_smile: https://github.com/conan-io/conan-center-index/pull/2729

**Comment by keysight-daryl** [comment]
I think it would be good idea to add the upstream paho-mqtt-c author @icraggs on all PR reviews for this package since he is now active on here developing this.

**Comment by SpaceIm** [comment]
I close this PR in favor of https://github.com/conan-io/conan-center-index/pull/2739

**Pull request closed without merging**